### PR TITLE
log error in messages.log after 401

### PR DESCRIPTION
**Pull request by jimmy1wu**
for https://github.com/OpenLiberty/open-liberty/issues/17613

### CODE REVIEW

**Comment by jimmy1wu** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.jwt_fat.consumer,com.ibm.ws.security.jwt_fat.builder,com.ibm.ws.security.jwtsso_fat,com.ibm.ws.security.mp.jwt.1.1_fat,com.ibm.ws.security.mp.jwt.1.2_fat,com.ibm.ws.security.oidc.client_fat.1,com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.config,com.ibm.ws.security.oidc.client_fat.e2e,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.saml,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.app-passwords,com.ibm.ws.security.oidc.server_fat.app-passwords.DerbyDB.jwt,com.ibm.ws.security.oidc.server_fat.app-passwords.DerbyDB.opaque,com.ibm.ws.security.oidc.server_fat.app-passwords.LocalStore.jwt,com.ibm.ws.security.oidc.server_fat.app-passwords.LocalStore.opaque,com.ibm.ws.security.oidc.server_fat.app-tokens,com.ibm.ws.security.oidc.server_fat.app-tokens.DerbyDB.jwt,com.ibm.ws.security.oidc.server_fat.app-tokens.DerbyDB.opaque,com.ibm.ws.security.oidc.server_fat.app-tokens.LocalStore.jwt,com.ibm.ws.security.oidc.server_fat.app-tokens.LocalStore.opaque,com.ibm.ws.security.oidc.server_fat.clientregistration,com.ibm.ws.security.oidc.server_fat.config,com.ibm.ws.security.oidc.server_fat.config.oauth,com.ibm.ws.security.oidc.server_fat.config.oidc,com.ibm.ws.security.oidc.server_fat.endpoint.authorize,com.ibm.ws.security.oidc.server_fat.endpoint.coveragemap,com.ibm.ws.security.oidc.server_fat.endpoint.discovery,com.ibm.ws.security.oidc.server_fat.endpoint.end_session,com.ibm.ws.security.oidc.server_fat.endpoint.introspect,com.ibm.ws.security.oidc.server_fat.endpoint.revoke,com.ibm.ws.security.oidc.server_fat.endpoint.token,com.ibm.ws.security.oidc.server_fat.endpoint.userinfo,com.ibm.ws.security.oidc.server_fat.feature,com.ibm.ws.security.oidc.server_fat.granttype.jwt,com.ibm.ws.security.oidc.server_fat.gui.tokenMgmt,com.ibm.ws.security.oidc.server_fat.jaxrs,com.ibm.ws.security.oidc.server_fat.jaxrs.config,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.saml,com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.LibertyOP,com.ibm.ws.security.social_fat.OpenShift,com.ibm.ws.security.social_fat.delegated,com.ibm.ws.security.social_fat.multiProvider,com.ibm.ws.security.social_fat.okdServiceLogin

File: dev/com.ibm.ws.security.openidconnect.clients.common/resources/com/ibm/ws/security/openidconnect/clients/common/resources/OidcClientMessages.nlsprops [hunk 1]
```diff
@@ -207,4 +208,14 @@ USERINFO_INVALID=CWWKS1749E: The User Info data [{0}] is invalid because the sub
 USERINFO_INVALID.explanation=The sub claim of user info data is required to match the sub claim of the ID token, but it does not. 
 USERINFO_INVALID.useraction=Ensure that the OpenID Connect provider generates valid User Info data.
 
-# STOP HERE, OIDC COMMON HAS RESERVED 1751 - 1799 
\ No newline at end of file
+# STOP HERE, OIDC COMMON HAS RESERVED 1751 - 1799 
+
+OIDC_CLIENT_NULL_AUTH_ENDPOINT=CWWKS1534E: The OpenID Connect client [{0}] requires an authorization endpoint URL, but it is not set.
+OIDC_CLIENT_NULL_AUTH_ENDPOINT.explanation=An authorization endpoint URL must be set in order to obtain an authorization code or the required tokens.
```
Comment (helyarp) [inline dev/com.ibm.ws.security.openidconnect.clients.common/resources/com/ibm/ws/security/openidconnect/clients/common/resources/OidcClientMessages.nlsprops]: Remove "in order" (change "in order to" to "to").

File: dev/com.ibm.ws.security.openidconnect.clients.common/resources/com/ibm/ws/security/openidconnect/clients/common/resources/OidcClientMessages.nlsprops [hunk 2]
```diff
@@ -207,4 +208,14 @@ USERINFO_INVALID=CWWKS1749E: The User Info data [{0}] is invalid because the sub
...
+OIDC_CLIENT_NULL_AUTH_ENDPOINT.useraction=Set the authorizationEndpointUrl attribute in the OpenID Connect client configuration to the authorization endpoint URL of the OpenID Connect provider. Alternatively, set the discoveryEndpointUrl attribute in the OpenID Connect client configuration to the discovery endpoint URL of the OpenID Connect provider.
+
+OIDC_CLIENT_NULL_TOKEN_ENDPOINT=CWWKS1535E: The OpenID Connect client [{0}] requires a token endpoint URL, but it is not set.
+OIDC_CLIENT_NULL_TOKEN_ENDPOINT.explanation=A token endpoint URL must be set in order to obtain the required tokens.
```
Comment (helyarp) [inline dev/com.ibm.ws.security.openidconnect.clients.common/resources/com/ibm/ws/security/openidconnect/clients/common/resources/OidcClientMessages.nlsprops]: Remove "in order"

**Review by helyarp — COMMENTED** [review]
Some suggested changes.

**Comment by skasund** [comment]
L2 message review completed

**Comment by jimmy1wu** [comment]
#libby

**Comment by jimmy1wu** [comment]
@helyarp thanks for the review
PTAL, i have addressed your comments

**Comment by jimmy1wu** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.jwt_fat.consumer,com.ibm.ws.security.jwt_fat.builder,com.ibm.ws.security.jwtsso_fat,com.ibm.ws.security.mp.jwt.1.1_fat,com.ibm.ws.security.mp.jwt.1.2_fat,com.ibm.ws.security.oidc.client_fat.1,com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.config,com.ibm.ws.security.oidc.client_fat.e2e,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.saml,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.app-passwords,com.ibm.ws.security.oidc.server_fat.app-passwords.DerbyDB.jwt,com.ibm.ws.security.oidc.server_fat.app-passwords.DerbyDB.opaque,com.ibm.ws.security.oidc.server_fat.app-passwords.LocalStore.jwt,com.ibm.ws.security.oidc.server_fat.app-passwords.LocalStore.opaque,com.ibm.ws.security.oidc.server_fat.app-tokens,com.ibm.ws.security.oidc.server_fat.app-tokens.DerbyDB.jwt,com.ibm.ws.security.oidc.server_fat.app-tokens.DerbyDB.opaque,com.ibm.ws.security.oidc.server_fat.app-tokens.LocalStore.jwt,com.ibm.ws.security.oidc.server_fat.app-tokens.LocalStore.opaque,com.ibm.ws.security.oidc.server_fat.clientregistration,com.ibm.ws.security.oidc.server_fat.config,com.ibm.ws.security.oidc.server_fat.config.oauth,com.ibm.ws.security.oidc.server_fat.config.oidc,com.ibm.ws.security.oidc.server_fat.endpoint.authorize,com.ibm.ws.security.oidc.server_fat.endpoint.coveragemap,com.ibm.ws.security.oidc.server_fat.endpoint.discovery,com.ibm.ws.security.oidc.server_fat.endpoint.end_session,com.ibm.ws.security.oidc.server_fat.endpoint.introspect,com.ibm.ws.security.oidc.server_fat.endpoint.revoke,com.ibm.ws.security.oidc.server_fat.endpoint.token,com.ibm.ws.security.oidc.server_fat.endpoint.userinfo,com.ibm.ws.security.oidc.server_fat.feature,com.ibm.ws.security.oidc.server_fat.granttype.jwt,com.ibm.ws.security.oidc.server_fat.gui.tokenMgmt,com.ibm.ws.security.oidc.server_fat.jaxrs,com.ibm.ws.security.oidc.server_fat.jaxrs.config,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.saml,com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.LibertyOP,com.ibm.ws.security.social_fat.OpenShift,com.ibm.ws.security.social_fat.delegated,com.ibm.ws.security.social_fat.multiProvider,com.ibm.ws.security.social_fat.okdServiceLogin

**Review by helyarp — COMMENTED** [review]
Looks good!

**Review by ayoho — APPROVED** [review]

**Comment by jimmy1wu** [comment]
#libby

**Comment by jimmy1wu** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.jwt_fat.consumer,com.ibm.ws.security.jwt_fat.builder,com.ibm.ws.security.jwtsso_fat,com.ibm.ws.security.mp.jwt.1.1_fat,com.ibm.ws.security.mp.jwt.1.2_fat,com.ibm.ws.security.oidc.client_fat.1,com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.config,com.ibm.ws.security.oidc.client_fat.e2e,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.saml,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.app-passwords,com.ibm.ws.security.oidc.server_fat.app-passwords.DerbyDB.jwt,com.ibm.ws.security.oidc.server_fat.app-passwords.DerbyDB.opaque,com.ibm.ws.security.oidc.server_fat.app-passwords.LocalStore.jwt,com.ibm.ws.security.oidc.server_fat.app-passwords.LocalStore.opaque,com.ibm.ws.security.oidc.server_fat.app-tokens,com.ibm.ws.security.oidc.server_fat.app-tokens.DerbyDB.jwt,com.ibm.ws.security.oidc.server_fat.app-tokens.DerbyDB.opaque,com.ibm.ws.security.oidc.server_fat.app-tokens.LocalStore.jwt,com.ibm.ws.security.oidc.server_fat.app-tokens.LocalStore.opaque,com.ibm.ws.security.oidc.server_fat.clientregistration,com.ibm.ws.security.oidc.server_fat.config,com.ibm.ws.security.oidc.server_fat.config.oauth,com.ibm.ws.security.oidc.server_fat.config.oidc,com.ibm.ws.security.oidc.server_fat.endpoint.authorize,com.ibm.ws.security.oidc.server_fat.endpoint.coveragemap,com.ibm.ws.security.oidc.server_fat.endpoint.discovery,com.ibm.ws.security.oidc.server_fat.endpoint.end_session,com.ibm.ws.security.oidc.server_fat.endpoint.introspect,com.ibm.ws.security.oidc.server_fat.endpoint.revoke,com.ibm.ws.security.oidc.server_fat.endpoint.token,com.ibm.ws.security.oidc.server_fat.endpoint.userinfo,com.ibm.ws.security.oidc.server_fat.feature,com.ibm.ws.security.oidc.server_fat.granttype.jwt,com.ibm.ws.security.oidc.server_fat.gui.tokenMgmt,com.ibm.ws.security.oidc.server_fat.jaxrs,com.ibm.ws.security.oidc.server_fat.jaxrs.config,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.saml,com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.LibertyOP,com.ibm.ws.security.social_fat.OpenShift,com.ibm.ws.security.social_fat.delegated,com.ibm.ws.security.social_fat.multiProvider,com.ibm.ws.security.social_fat.okdServiceLogin

**Comment by jimmy1wu** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.jwt_fat.consumer,com.ibm.ws.security.jwt_fat.builder,com.ibm.ws.security.jwtsso_fat,com.ibm.ws.security.mp.jwt.1.1_fat,com.ibm.ws.security.mp.jwt.1.2_fat,com.ibm.ws.security.oidc.client_fat.1,com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.config,com.ibm.ws.security.oidc.client_fat.e2e,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.saml,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.app-passwords,com.ibm.ws.security.oidc.server_fat.app-passwords.DerbyDB.jwt,com.ibm.ws.security.oidc.server_fat.app-passwords.DerbyDB.opaque,com.ibm.ws.security.oidc.server_fat.app-passwords.LocalStore.jwt,com.ibm.ws.security.oidc.server_fat.app-passwords.LocalStore.opaque,com.ibm.ws.security.oidc.server_fat.app-tokens,com.ibm.ws.security.oidc.server_fat.app-tokens.DerbyDB.jwt,com.ibm.ws.security.oidc.server_fat.app-tokens.DerbyDB.opaque,com.ibm.ws.security.oidc.server_fat.app-tokens.LocalStore.jwt,com.ibm.ws.security.oidc.server_fat.app-tokens.LocalStore.opaque,com.ibm.ws.security.oidc.server_fat.clientregistration,com.ibm.ws.security.oidc.server_fat.config,com.ibm.ws.security.oidc.server_fat.config.oauth,com.ibm.ws.security.oidc.server_fat.config.oidc,com.ibm.ws.security.oidc.server_fat.endpoint.authorize,com.ibm.ws.security.oidc.server_fat.endpoint.coveragemap,com.ibm.ws.security.oidc.server_fat.endpoint.discovery,com.ibm.ws.security.oidc.server_fat.endpoint.end_session,com.ibm.ws.security.oidc.server_fat.endpoint.introspect,com.ibm.ws.security.oidc.server_fat.endpoint.revoke,com.ibm.ws.security.oidc.server_fat.endpoint.token,com.ibm.ws.security.oidc.server_fat.endpoint.userinfo,com.ibm.ws.security.oidc.server_fat.feature,com.ibm.ws.security.oidc.server_fat.granttype.jwt,com.ibm.ws.security.oidc.server_fat.gui.tokenMgmt,com.ibm.ws.security.oidc.server_fat.jaxrs,com.ibm.ws.security.oidc.server_fat.jaxrs.config,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.saml,com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.LibertyOP,com.ibm.ws.security.social_fat.OpenShift,com.ibm.ws.security.social_fat.delegated,com.ibm.ws.security.social_fat.multiProvider,com.ibm.ws.security.social_fat.okdServiceLogin

**Pull request merged**